Samples/WinMLSamplesGallery/WinMLSamplesGalleryNative/SegmentModel.h

```diff
@@ -0,0 +1,137 @@
+LearningModel ReshapeFlatBufferToNCHW(long n, long c, long h, long w);
+LearningModel Invert(long n, long c, long h, long w);
+
+class StreamModelBase
+{
+public:
+	StreamModelBase() :
+		m_inputVideoFrame(nullptr),
+		m_outputVideoFrame(nullptr),
+		m_session(nullptr),
+		m_binding(nullptr),
+		m_syncStarted(false) {}
+
+	virtual ~StreamModelBase() {
+		if(m_session) m_session.Close();
+		if(m_binding) m_binding.Clear();
+		if (m_inputVideoFrame) m_inputVideoFrame.Close();
+		if (m_outputVideoFrame) m_outputVideoFrame.Close(); 
+	};
+
+	virtual void InitializeSession(int w, int h) = 0;
+	virtual void Run(IDirect3DSurface src, IDirect3DSurface dest) = 0;
+	
+	// Synchronous eval status
+	bool m_syncStarted = false; 
+	VideoFrame m_outputVideoFrame;
+	static const int m_scale = 4;
+	winrt::hstring m_modelBasePath;
+
+protected:
+	// Cache input frames into a shareable d3d-backed VideoFrame
+	void SetVideoFrames(VideoFrame inVideoFrame, VideoFrame outVideoFrame) 
+	{
+		if (true || !m_videoFramesSet)
+		{
+			auto device = m_session.Device().Direct3D11Device();
+			auto inDesc = inVideoFrame.Direct3DSurface().Description();
+			auto outDesc = outVideoFrame.Direct3DSurface().Description();
+			/*
+				NOTE: VideoFrame::CreateAsDirect3D11SurfaceBacked takes arguments in (width, height) order
+				whereas every model created with LearningModelBuilder takes arguments in (height, width) order. 
+			*/ 
+			auto format = winrt::Windows::Graphics::DirectX::DirectXPixelFormat::B8G8R8X8UIntNormalized;
+			m_inputVideoFrame = VideoFrame::CreateAsDirect3D11SurfaceBacked(format, m_imageWidthInPixels, m_imageHeightInPixels, device);
+			m_outputVideoFrame = VideoFrame::CreateAsDirect3D11SurfaceBacked(format, m_imageWidthInPixels, m_imageHeightInPixels, device);
+			m_videoFramesSet = true;
+		}
+		// NOTE: WinML supports mainly RGB-formatted video frames, which aren't backed by a shareable surface by the Capture Engine. 
+		// Copying to a new VideoFrame makes it shareable for use in inference. 
+		inVideoFrame.CopyToAsync(m_inputVideoFrame).get();
+		outVideoFrame.CopyToAsync(m_outputVideoFrame).get();
+	}
+
+	void SetImageSize(int w, int h) {
+		m_imageWidthInPixels = w;
+		m_imageHeightInPixels = h;
+	}
+	
+	LearningModelSession CreateLearningModelSession(const LearningModel& model, bool closedModel = true) {
+		auto device = LearningModelDevice(m_useGPU ? LearningModelDeviceKind::DirectXHighPerformance : LearningModelDeviceKind::Default);
+		auto options = LearningModelSessionOptions();
+		options.BatchSizeOverride(0);
+		options.CloseModelOnSessionCreation(closedModel);
+		auto session = LearningModelSession(model, device, options);
+		return session;
+	}
+
+	bool						m_useGPU = true;
+	bool						m_videoFramesSet = false;
+	VideoFrame					m_inputVideoFrame;
+								
+	UINT32                      m_imageWidthInPixels = 0;
+	UINT32                      m_imageHeightInPixels = 0;
+
```

Comment:
kill tabs

That might be what is causing your indent to be so far over, instead of just:

```
 bool        m_useGPU = true;
 bool        m_videoFramesSet = false;
 VideoFrame  m_inputVideoFrame;
        
 UINT32      m_imageWidthInPixels = 0;
 UINT32      m_imageHeightInPixels = 0;
```